agent/app/agent.go

```diff
@@ -232,6 +232,8 @@ func (agent *ecsAgent) newTaskEngine(containerChangeEventStream *eventstream.Eve
 		log.Warnf(instanceIDMismatchErrorFormat,
 			previousEC2InstanceID, currentEC2InstanceID)
 
+		// Reset agent state as a new container instance
+		state.Reset()
```

Comment:
I don't think so, the file will be override later. And if agent restart at this point, it will run into the same code path `state.Reset` will be executed again.